openverse_catalog/dags/providers/provider_api_scripts/wikimedia_commons.py

```diff
@@ -21,450 +21,428 @@
+        return None
 
-    funcs = {
-        IMAGE: _add_image,
-        AUDIO: _add_audio,
-    }
-    funcs[valid_mediatype](parsed_data, media_data, media_info)
-
-
-def _get_value_by_name(key_value_list: list, prop_name: str):
-    prop_list = [
-        key_value_pair
-        for key_value_pair in key_value_list
-        if key_value_pair["name"] == prop_name
-    ]
-    if prop_list:
-        return prop_list[0].get("value")
-
-
-def _get_value_by_names(key_value_list: list, prop_names: list):
-    """Gets the first available value for one of the `prop_names`
-    property names"""
-    for prop_name in prop_names:
-        if val := _get_value_by_name(key_value_list, prop_name):
-            return val
+    def get_media_pages(self, response_json):
+        if response_json is not None:
+            image_pages = response_json.get("query", {}).get("pages")
+            if image_pages is not None:
+                logger.info(f"Got {len(image_pages)} pages")
+                return image_pages
 
+        logger.warning(f"No pages in the image batch: {response_json}")
+        return None
 
-def _parse_audio_file_data(parsed_data: dict, file_metadata: list) -> dict:
-    streams = _get_value_by_name(file_metadata, "streams")
-    try:
+    def get_record_data(self, record):
+        foreign_id = record.get("pageid")
+        logger.debug(f"Processing page ID: {foreign_id}")
+
+        media_info = self.get_media_info_dict(record)
+
+        valid_media_type = self.extract_media_type(media_info)
+        if not valid_media_type:
+            # Do not process unsupported media types, like Video
+            return None
+
+        license_info = self.get_license_info(media_info)
+        if license_info.url is None:
+            return None
+
+        media_url = media_info.get("url")
+        if media_url is None:
+            return None
+
+        creator, creator_url = self.extract_creator_info(media_info)
+        title = self.extract_title(media_info)
+        filesize = media_info.get("size", 0)  # in bytes
+        filetype = self.extract_file_type(media_info)
+        meta_data = self.create_meta_data_dict(record)
+
+        record_data = {
+            "media_url": media_url,
+            "foreign_landing_url": media_info.get("descriptionshorturl"),
+            "foreign_identifier": foreign_id,
+            "license_info": license_info,
+            "creator": creator,
+            "creator_url": creator_url,
+            "title": title,
+            "filetype": filetype,
+            "filesize": filesize,
+            "meta_data": meta_data,
+            "media_type": valid_media_type,
+        }
+
+        # Extend record_data with media-type specific fields
+        funcs = {
+            IMAGE: self.get_image_record_data,
+            AUDIO: self.get_audio_record_data,
+        }
+        return funcs[valid_media_type](record_data, media_info)
+
+    def get_image_record_data(self, record_data, media_info):
```

Comment:
You're right! I think I was avoiding it just because most of them use `extract_ext_value` and accessing it statically is a bit clunky looking, but we probably should :)